src/lib/transition_frontier_controller_tests/test_transition_frontier.ml

```diff
@@ -26,6 +27,38 @@ let%test_module "Root_history and Transition_frontier" =
     let breadcrumb_trail_equals =
       List.equal ~equal:Transition_frontier.Breadcrumb.equal
 
+    let%test "Common ancestor should return the youngest ancestor of two \
+              breadcrumbs" =
+      let logger = Logger.null () in
+      Async.Thread_safe.block_on_async_exn (fun () ->
+          let%bind frontier = create_root_frontier ~logger in
+          let root = Transition_frontier.root frontier in
+          let%bind ancestors =
+            create_breadcrumbs ~logger
+              ~size:(max_length - (max_length / 2))
+              root
+          in
+          let youngest_ancestor = List.last_exn ancestors in
+          let%bind () =
+            Deferred.List.iter ancestors ~f:(fun ancestor ->
+                Transition_frontier.add_breadcrumb_exn frontier ancestor )
+          in
+          let%bind branch1 =
+            create_breadcrumbs ~logger ~size:(max_length / 2) youngest_ancestor
```

Comment:
Small nit, can you randomly generate a size between 1 and max_legnth/2  for both branches? I want to show in the test that we can retrieve the breadcrumbs successfully even with two uneven branches.

Reply:
Sure, I would do that.